src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
```

Comment:
Honestly, if PHP could autoload functions, I'd prefer it be a function. [There's absolutely no reason to be beholden to objects](http://steve-yegge.blogspot.com/2006/03/execution-in-kingdom-of-nouns.html) :)